test/test_cuda_expandable_segments.py

```diff
@@ -2,14 +2,36 @@
 # run time cuda tests, but with the allocator using expandable segments
 
 import os
+import pathlib
+import subprocess
+import sys
 
 import torch
 
-from torch.testing._internal.common_cuda import IS_JETSON
+from torch.testing._internal.common_cuda import IS_JETSON, IS_WINDOWS
+
+REPO_ROOT = pathlib.Path(__file__).resolve().parent.parent
+
+sys.path.insert(0, str(REPO_ROOT))
+from tools.stats.import_test_stats import get_disabled_tests
+
+if torch.cuda.is_available() and not IS_JETSON and not IS_WINDOWS:
+    get_disabled_tests(".")
+
+    env = os.environ.copy()
+    env["PYTORCH_CUDA_ALLOC_CONF"] = "expandable_segments:True"
+    env["DISABLED_TESTS_FILE"] = ".pytorch-disabled-tests.json"
 
-if torch.cuda.is_available() and not IS_JETSON:
     torch.cuda.memory._set_allocator_settings("expandable_segments:True")
 
     current_dir = os.path.dirname(os.path.abspath(__file__))
-    filepath = os.path.join(current_dir, "test_cuda.py")
-    exec(compile(open(filepath).read(), filepath, mode="exec"))
+
+    test_cuda_filepath = os.path.join(current_dir, "test_cuda.py")
+    test_cudagraphs_filepath = os.path.join(current_dir, "dynamo/test_cudagraphs.py")
+    test_cudagraph_trees_filepath = os.path.join(
+        current_dir, "inductor/test_cudagraph_trees.py"
+    )
+
+    subprocess.run([sys.executable, test_cuda_filepath], env=env, check=True)
```

Comment:
The downside of doing it this way is that it no longer works to pass individual test flags to this file, or use pytest to run it. Would it be possible to just make new files for the two added test harnesses?